jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPart.java

```diff
@@ -383,13 +433,14 @@ public void close()
                 closed = true;
                 if (!contentSources.isEmpty())
                 {
-                    contentSourcesToFail = new ArrayList<>(contentSources);
+                    contentSourcesToFail = List.copyOf(contentSources);
                     contentSources.clear();
                 }
+                content.forEach(Content.Chunk::release);
+                content.clear();
             }
 
             super.close();
```

Comment:
Calling `super` calls `fail()`, which will do nothing because the data structures have been cleared.
I find this confusing (as you think "am I failing twice?"), so perhaps just don't call `super` at all, as we are doing all the work anyway in this overridden version.

Reply:
Actually, `fail()` and `close()` are almost doing the same thing, and since `super.close()` already calls `fail()`, `close()` can just go away.